### DISCUSSION TITLE
502: problem with authelia-authrequest.conf

### DISCUSSION THREAD

**Discussion by daerlnaxe** [authelia/authelia#6483] [Troubleshooting] [answered]
Hi,

I found a problem with the authelia-authrequest.conf file, with this code (from official documentation)
```nginx
set $upstream_authelia http://authelia:9091/api/verify;

## Virtual endpoint created by nginx to forward auth requests.
location = /authelia {
    ## Essential Proxy Configuration
    internal;
    proxy_pass $upstream_authelia;
# ...
}
```
NGINX 1.20.1

I don't know why but i get a 502 error code.

if i set the proxy_pass to the same address instead of using a variable, it works.

**Comment by james-d-elliott** [comment]
The nginx and authelia logs will reveal why. There's not a specific issue with the file though, it's just something you've configured differently or incorrectly to the assumptions laid out in the preamble.

**Reply by daerlnaxe** [reply]
Thx for your answer but ... how ?

I mean, I'm ok for all you want but i don't understand how it can be possible. There is a variable declared just above `set $upstream_authelia http://authelia:9091/api/verify;`
And just after there is a usage without anything between . 

As i said i just replaced the value by the one used to initialize the variable. I add, even if move to glue just above the usage i have the same problem. I ask myself if there is not a problem with a nginx version after all it's +/-2 years old if i read releases date.

I tried to replace the name for the variable by raoul (i didn't paste it, it's not really glorious, but i would to remove special chars just to see) and problem remains.

In all cases it was just to let a message if someone has the same problem instead to be stuck .

Best regards.

**Reply by james-d-elliott** [reply]
> Thx for your answer but ... how ?

Show the requested logs and maybe we can tell you. Show the authelia logs from starting the application to the issue occurring, likewise with nginx.

> As i said i just replaced the value by the one used to initialize the variable. I add, even if move to glue just above the usage i have the same problem. I ask myself if there is not a problem with a nginx version after all it's +/-2 years old if i read releases date.

Please read what I said again, I never said the issue specifically with nginx. The issue is likely with *your configuration* of nginx based upon decisions you've not accounted for in the adjustment of the configuration of either nginx or other services like docker. 

However it should be noted that a version 3 years old is no longer a supported version by nginx (they ceased all support for it at the start of 2022) and there are about 10 disclosed security vulnerabilities with a version that old.

This configuration is heavily tested. We have thousands of users of it and we perform automated integration testing with it every time code is merged.

**Reply by james-d-elliott** [reply, marked as answer]
It should also be noted, that this is **_not the correct content_** for that file name. It should be per below (with the only alteration being `https://auth.example.com` replaced with your authelia domain and potentially the `set_escape_uri` vs `set` change):

```nginx
## Send a subrequest to Authelia to verify if the user is authenticated and has permission to access the resource.
auth_request /authelia;

## Set the $target_url variable based on the original request.

## Comment this line if you're using nginx without the http_set_misc module.
set_escape_uri $target_url $scheme://$http_host$request_uri;

## Uncomment this line if you're using NGINX without the http_set_misc module.
# set $target_url $scheme://$http_host$request_uri;

## Save the upstream response headers from Authelia to variables.
auth_request_set $user $upstream_http_remote_user;
auth_request_set $groups $upstream_http_remote_groups;
auth_request_set $name $upstream_http_remote_name;
auth_request_set $email $upstream_http_remote_email;

## Inject the response headers from the variables into the request made to the backend.
proxy_set_header Remote-User $user;
proxy_set_header Remote-Groups $groups;
proxy_set_header Remote-Name $name;
proxy_set_header Remote-Email $email;

## If the subreqest returns 200 pass to the backend, if the subrequest returns 401 redirect to the portal.
error_page 401 =302 https://auth.example.com/?rd=$target_url;
```
